modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java

```diff
@@ -77,4 +87,50 @@ private Method getCreateArrayOfMethod(Connection delegate)
 
         return createArrayMethod;
     }
+
+    @Override
+    public Struct createStruct(String typeName, Object[] attributes) throws SQLException
+    {
+        return super.createStruct(typeName, attributes);
+    }
+
+    @Override
+    protected void resolveLobs(String typeName, Object[] attributes) throws SQLException
+    {
+        Map<Integer, String> dataTypes = getDataTypes(typeName);
+
+        for (int index : dataTypes.keySet())
+        {
+            String dataTypeName = dataTypes.get(index);
+            doResolveLobIn(attributes, index-1, dataTypeName);
+        }
+    }
+
+    private Map<Integer, String> getDataTypes(String typeName) throws SQLException
+    {
+        Map<Integer, String> dataTypes = new HashMap<Integer, String>();
+
+        PreparedStatement ps = this.prepareStatement(QUERY_TYPE_ATTRS);
+        ps.setString(1, typeName);
+        ResultSet resultSet = null;
+
+        try
+        {
+            resultSet = ps.executeQuery();
```

Comment:
doesnt ResultSet implement AutoClosable/Closable?

Reply:
It is not needed from java 7.